### PR TITLE
add tier-0 test case

### PR DESCRIPTION
**Pull request by pengshanyu**
We have several test cases for FFI, but we are missing some basic/simple sanity tests for qm, if it is installed properly and/or if environment started successfully. 
This is a draft version. Since the naming of the test files and test methods are simple, I hope to use this version as a starting point for improving the sanity test.
We would like to name this test plan tier-0, but need to rename the current tier-0 first, so I temporarily named it tier-2

resolve https://github.com/containers/qm/issues/342

### CODE REVIEW

File: plans/e2e/tier-2.fmf
Comment (Yarboa) [inline plans/e2e/tier-2.fmf:32]: I think this part could move to ./set-ffi-env-e2e script

File: plans/e2e/tier-2.fmf
Comment (Yarboa) [inline plans/e2e/tier-2.fmf:14]: SET_CS_REPO: yes
  SKIP_EPEL_REPO: yes
  
  Are not related to this repo

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [inline tests/qm-sanity-test/test.sh:77]: Consider checking the following
check_qm_podman_ok
`podman exec qm bash -c "podman info"` 
Completed exit 0

check_bluechi_controller_ok
`systemctl is-active bluechi-controller`

I would also add check that /var is on separate partition

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [inline tests/qm-sanity-test/test.sh:41]: Consider renaming check_qm_podman_quadlet_ok

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [inline tests/qm-sanity-test/test.sh:19]: It is better to check that bluechictl is not broken
Advise to use `bluechictl status` inside the host.
should return two nodes connected 

Suggestion, 
check_bluechi_is_ok
see my comment below
systemctl is-active bluechi-controller
systemctl status show 
It can replace the checks inside the method

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [inline tests/qm-sanity-test/test.sh:68]: I would use from the host lsblk or fdisk to identify partition for /var /
fdisk -l -o Device,Size,Type

verify host /var is on different partition
Difference between c9s and ostree in out put exist

**Review by Yarboa — COMMENTED** [review]
Please add the tests to .packit.yaml
to be visible on github tests

File: plans/e2e/tier-2.fmf
Comment (pengshanyu) [reply plans/e2e/tier-2.fmf:32]: Hi @Yarboa , thanks for the review.
Should this part be deleted or should it be kept here and added it to `./set-ffi-env-e2e` as well?

Comment (Yarboa) [reply plans/e2e/tier-2.fmf:32]: MHO, it should not be here, @dougsland WDYT?

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh:68]: What does the output of `lsblk` in ostree look like? 
I tried with a ostree image "auto-osbuild-qemu-rhivos9-minimal-ostree-x86_64-8657563.d519d68e.qcow2”, but qm is inactive on this image. So I don't know if the output in the following image is correct:
image

File: plans/e2e/tier-2.fmf
Comment (pengshanyu) [reply plans/e2e/tier-2.fmf]: removed

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh:104]: done

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh]: done

**Comment by pengshanyu** [comment]
> Please add the tests to .packit.yaml to be visible on github tests
Should I do it now in this PR or should I wait until this PR is merged before doing it?

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [reply tests/qm-sanity-test/test.sh]: here, you can see that there is not separate partition for /var
see this 
```
[root@8e2b5d13-97aa-458d-a321-7533436e4e2c ~]# lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0         7:0    0  12.7M  1 loop 
nvme0n1     259:0    0   250G  0 disk 
├─nvme0n1p1 259:1    0   100M  0 part /boot/efi
├─nvme0n1p2 259:2    0   300M  0 part /boot
├─nvme0n1p3 259:3    0   2.4G  0 part /var
└─nvme0n1p4 259:4    0 247.2G  0 part /sysroot/ostree/deploy/rhivos/var
                                      /etc
                                      /sysroot

```

Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh]: @Yarboa thank you for sharing.

**Comment by Yarboa** [comment]
> > Please add the tests to .packit.yaml to be visible on github tests
> 
> Should I do it now in this PR or should I wait until this PR is merged before doing it?

MHO, if this new packit test introduced, why not let it run, this way we confirm that test merged are working

Now, i see you calked the tests tier-2
I think that current tier-0 tests should be renamed to a different name, like 
multi-bluchi-agents, that is the real purpose of it, maybe we will remove it from tests later on.

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [inline tests/qm-sanity-test/test.sh:95]: some thing like this 
[[ "$(lsblk -o 'MAJ:MIN,TYPE,MOUNTPOINTS')" =~ "part /var" ]]

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh]: done

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh]: done

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh]: done

**Comment by pengshanyu** [comment]
> MHO, if this new packit test introduced, why not let it run, this way we confirm that test merged are working
> 
> Now, i see you calked the tests tier-2 I think that current tier-0 tests should be renamed to a different name, like multi-bluchi-agents, that is the real purpose of it, maybe we will remove it from tests later on.

Hi @Yarboa, I renamed current tier-0 tests to `multi-bluechi-agents`, but there are also tier-0 tests in `qm-connectivity` and `qm-unit-files`, how should they be renamed or do you want them to be tier-0 along with the sanity tests?

**Comment by Yarboa** [comment]
> > MHO, if this new packit test introduced, why not let it run, this way we confirm that test merged are working
> > Now, i see you calked the tests tier-2 I think that current tier-0 tests should be renamed to a different name, like multi-bluchi-agents, that is the real purpose of it, maybe we will remove it from tests later on.
> 
> Hi @Yarboa, I renamed current tier-0 tests to `multi-bluechi-agents`, but there are also tier-0 tests in `qm-connectivity` and `qm-unit-files`, how should they be renamed or do you want them to be tier-0 along with the sanity tests?

Please update the test of tier-0 similar to ffi tests
Please use this one
https://github.com/containers/qm/blob/main/.packit.yaml#L75

Please as you mentioned rename tier-0 tests as mentioned
and please enable old tier-0 for manual trigger only as per this 
https://packit.dev/posts/manual-triggering#manual-only-triggering-of-jobs

**Comment by pengshanyu** [comment]
@Yarboa thank for the reply.
`qm-connectivity ` run with `run-test-e2e`
the new tier-0 test run with `set-ffi-env-e2e`
they are different test plan, I'm not sure how to run them both as tier-0 at one tmt run

File: .packit.yaml
Comment (Yarboa) [inline .packit.yaml:96]: I think you can change identifier to qm-tier-0

File: tests/qm-sanity-test/main.fmf
Comment (Yarboa) [inline tests/qm-sanity-test/main.fmf:1]: missing polarion-id

File: tests/qm-sanity-test/test.sh
Comment (Yarboa) [inline tests/qm-sanity-test/test.sh:7]: In 2nd thought i think eacho one of the function in a test itself I think you could split 
tests/qm-sanity-test/main.fmf

Into smaller parts like 
/check-qm-up
test: source test.sh; check_qm_is_up

each one of the tests will have its own id
so in that case this script has 6 tests, if one of them is failing, all of the 6 are marked as failed

File: tests/qm-sanity-test/main.fmf
Comment (pengshanyu) [reply tests/qm-sanity-test/main.fmf:1]: I split the sanity test into 6 tests. Once confirm that each test description is OK, I will import them into Polarion to generate Polarion id.

File: .packit.yaml
Comment (pengshanyu) [reply .packit.yaml]: done

File: tests/qm-sanity-test/test.sh
Comment (pengshanyu) [reply tests/qm-sanity-test/test.sh]: @Yarboa thank you so much. I split the test into 6 tests, which seems more reasonable.

File: tests/qm-sanity-test/README.md
Comment (Yarboa) [inline tests/qm-sanity-test/README.md:13]: Please change all ASIL -> host 
We do not want to mislead people that this setup is FuSA certified

File: .packit.yaml
Comment (Yarboa) [inline .packit.yaml:60]: Can we run this test on demand? 
not on each PR? please take a look here 
https://packit.dev/posts/manual-triggering#manual-only-triggering-of-jobs

Add the following 
```
  skip_build: true
  manual_trigger: true
```

File: plans/e2e/multi-bluechi-agents.fmf
Comment (Yarboa) [inline plans/e2e/multi-bluechi-agents.fmf:1]: isnt it a duplication of of plans ffi ? 
maybe you could use one plan for prepare could go under plans? and then the other plans for ffi and e2e will have their own difference in prepare step?

File: plans/e2e/tier-0.fmf
Comment (Yarboa) [inline plans/e2e/tier-0.fmf:14]: same go for here
All the common parts should go to main.fmf under plans

**Review by Yarboa — COMMENTED** [review]
Very nice work @pengshanyu 
Please see the comments above, 
I would not choose to number the tests, but it is up to you

File: tests/qm-sanity-test/test.sh
Comment (pbrilla-rh) [reply tests/qm-sanity-test/test.sh]: yes I agree it will add little bit granularity

File: plans/e2e/multi-bluechi-agents.fmf
Comment (pengshanyu) [reply plans/e2e/multi-bluechi-agents.fmf:1]: I tried to extract the common parts of `ffi.fmf` and `tier-0.fmf ` to `plans/main.fmf ` , but main.fmf will affect the `multi-bluechi-agents.fmf` and `tier-1.fmf`. This makes me feel a bit confusing.

File: tests/qm-sanity-test/README.md
Comment (pengshanyu) [reply tests/qm-sanity-test/README.md]: done

File: .packit.yaml
Comment (pengshanyu) [reply .packit.yaml:60]: done

**Comment by Yarboa** [comment]
LGTM, 
@pengshanyu well done
Please squash all comments, to have clean commits history before merge

File: tests/qm-sanity-test/main.fmf
Comment (pengshanyu) [reply tests/qm-sanity-test/main.fmf]: added polarion-id

**Comment by pengshanyu** [comment]
@Yarboa, thank you so much for the review. I added the Polarion-id.

**Comment by pengshanyu** [comment]
Hi @Yarboa , after this PR merge, we should also exclude this sanity test cases in the p-l-p test plan, right?

File: plans/main.fmf
Comment (pengshanyu) [reply plans/main.fmf]: removed PACKIT_COPR_PROJECT and PACKAGE

**Comment by pengshanyu** [comment]
packit test --identifier e2e-multi-bluechi-agents

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**
